app/src/main/java/com/cornellappdev/android/eatery/WeeklyMenuFragment.java

```diff
@@ -0,0 +1,364 @@
+					}
+				});
+
+		dinnerText.setOnClickListener(
+				new View.OnClickListener() {
+					@Override
+					public void onClick(View view) {
+						breakfastText.setTextColor(ContextCompat.getColor(getActivity().getApplicationContext(), R.color.secondary));
+						lunchText.setTextColor(ContextCompat.getColor(getActivity().getApplicationContext(), R.color.secondary));
+						dinnerText.setTextColor(ContextCompat.getColor(getActivity().getApplicationContext(), R.color.blue));
+
+						mealType = MealType.DINNER;
+						changeListAdapter(mealType, presenter.getSelectedDate());
+					}
+				});
+		LocalDateTime currentTime = LocalDateTime.now();
+
+		if (currentTime.getHour() < 11) {
+			changeListAdapter(MealType.BREAKFAST, presenter.getSelectedDate());
+			mealType = MealType.BREAKFAST;
+			breakfastText.setTextColor(ContextCompat.getColor(getActivity().getApplicationContext(), R.color.blue));
+			lunchText.setTextColor(ContextCompat.getColor(getActivity().getApplicationContext(), R.color.secondary));
+			dinnerText.setTextColor(ContextCompat.getColor(getActivity().getApplicationContext(), R.color.secondary));
+		} else if (currentTime.getHour() < 16) {
+			changeListAdapter(MealType.LUNCH, presenter.getSelectedDate());
+			mealType = MealType.LUNCH;
+			breakfastText.setTextColor(ContextCompat.getColor(getActivity().getApplicationContext(), R.color.secondary));
+			lunchText.setTextColor(ContextCompat.getColor(getActivity().getApplicationContext(), R.color.blue));
+			dinnerText.setTextColor(ContextCompat.getColor(getActivity().getApplicationContext(), R.color.secondary));
+		} else if (currentTime.getHour() < 22) {
+			changeListAdapter(MealType.DINNER, presenter.getSelectedDate());
+			mealType = MealType.DINNER;
+			breakfastText.setTextColor(ContextCompat.getColor(getActivity().getApplicationContext(), R.color.secondary));
+			lunchText.setTextColor(ContextCompat.getColor(getActivity().getApplicationContext(), R.color.secondary));
+			dinnerText.setTextColor(ContextCompat.getColor(getActivity().getApplicationContext(), R.color.blue));
+		} else {
+			presenter.getDayInWeek(1);
+			changeListAdapter(MealType.BREAKFAST, presenter.getSelectedDate());
+			mealType = MealType.BREAKFAST;
+			breakfastText.setTextColor(ContextCompat.getColor(getActivity().getApplicationContext(), R.color.blue));
+			lunchText.setTextColor(ContextCompat.getColor(getActivity().getApplicationContext(), R.color.secondary));
+			dinnerText.setTextColor(ContextCompat.getColor(getActivity().getApplicationContext(), R.color.secondary));
+		}
+		return rootView;
+	}
+
+	/**
+	 * Changes the text color to black when selected
+	 */
+	@Override
+	public void onClick(View v) {
+		TextView tv = (TextView) v;
+
+		tv.setTextColor(ContextCompat.getColor(getActivity().getApplicationContext(), R.color.primary));
+		changeDateColor(tv);
+
+		int id = tv.getId();
+		LocalDate currentDate = presenter.getSelectedDate();
+
+		switch (id) {
```

Comment:
this seems pretty extraneous –– you're testing whether an `int` is equal to a value, and wrapping it in a `switch` statement, maybe something like this (pseudocode, feel free to come up with something better):

```
int[] example = {R.id.date0, ..., R.id.date6};
currentDate = presenter.getDayInWeek(example.indexOf(id));
```

Clearly this doesn't translate exactly, but there's definitely a cleaner way to do this in Java!

Reply:
wow this is genius!